scrapers/sc/bills.py

```diff
@@ -90,6 +90,51 @@ def action_type(action):
     return None
 
 
+def get_index_url(session, chamber, chamber_letter):
+    """
+    Note: a Web Archive ID should be added during session transitions to ensure
+    functionality of past session scrapes.
+    1. Search chamber-specific `index_url` on https://web.archive.org/
+    2. Select latest snapshot in the session end year that still has bill intros
+    listed for the full session (typically this is the early November snapshot)
+    3. The Web Archive ID is the 14-digit number string in that snapshot's URL
+
+    Example of 2019-2020 House bill index URL:
+    https://web.archive.org/web/20201101155143/https://www.scstatehouse.gov/sessphp/hintros.php
+    """
+    index_url = f"https://www.scstatehouse.gov/sessphp/{chamber_letter}intros.php"
+
+    # Dictionary of ids for URL paths to archived HTML of bill index pages
+    # from the prior two sessions. (Archives needed or index_url leads to current session bills.)
+    web_archive_ids = {
+        "2017-2018-lower": "20181101144929",
+        "2017-2018-upper": "20181101144422",
+        "2019-2020-lower": "20201101155143",
+        "2019-2020-upper": "20201101152857",
+        "2021-2022-lower": "20221110101038",
+        "2021-2022-upper": "20221110101352",
+        # TODO: Add archive id values during transition between sessions (i.e. Dec 2024).
+        "2023-2024-lower": None,
```

Comment:
I can see what you're doing by padding out future sessions, which works fine. To make it a little less verbose, you could instead have a little logic that checks to see if the key exists in the `web_archive_ids` dict, and then provides a default value of `None` if it's not present: https://stackoverflow.com/a/1602964

Something like:

```
web_archive_ids = {
        "2017-2018-lower": "20181101144929",
        "2017-2018-upper": "20181101144422",
        "2019-2020-lower": "20201101155143",
        "2019-2020-upper": "20201101152857",
        "2021-2022-lower": "20221110101038",
        "2021-2022-upper": "20221110101352",
}
web_archive_id = web_archive_ids.get(f"{session}-{chamber}", None)

Reply:
That's a really convenient optional parameter for the `get()` method that I completely forgot about. Thanks! Added in [49c5f72](https://github.com/openstates/openstates-scrapers/pull/4654/commits/49c5f722578b3298a2a71a4d1b7ed3f39dbde681).